src/main/java/org/cryptomator/common/LocationPreset.java

```diff
@@ -0,0 +1,60 @@
+package org.cryptomator.common;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Enum of common cloud providers and their default local storage location path.
+ */
+public enum LocationPreset {
+
+	DROPBOX("Dropbox", "~/Dropbox"),
+	ICLOUD("iCloud Drive", "~/Library/Mobile Documents/com~apple~CloudDocs", "~/iCloudDrive"),
+	GDRIVE("Google Drive", "~/Google Drive/My Drive", "~/Google Drive"),
+	MEGA("MEGA", "~/MEGA"),
+	ONEDRIVE("OneDrive", "~/OneDrive"),
+	PCLOUD("pCloud", "~/pCloudDrive"),
+
+	LOCAL("local");
+
+	final String name;
+	final List<Path> candidates;
+
+	LocationPreset(String name, String... candidates) {
+		this.name = name;
+
+		String userHome = System.getProperty("user.home");
```

Comment:
can be a constant, so that...

Reply:
If a constant (`static final`) is used, its value is always `null`. I don't know why, unfortunately. But this was the only way without quering the property for each enum value.

Reply:
Changed in [5ab74af](https://github.com/cryptomator/cryptomator/pull/2245/commits/5ab74affb5bb0dcb66858cb238608d3849acd3be). A explanation can be found in the inline comment for the inner class.